core/src/bytes/BytesReader.cpp

```diff
@@ -215,6 +211,54 @@ namespace ledger {
             }
         }
 
+        uint16_t BytesReader::readNextBeUint16() {
+            uint16_t result;
+            uint8_t* ptr = reinterpret_cast<uint8_t *>(&result);
+            for (auto i = 0; i < sizeof(result); i++) {
+                ptr[i] = readNextByte();
+            }
+            ledger::core::endianness::swapToEndianness(ptr, sizeof(result),
+                                                       ledger::core::endianness::Endianness::BIG,
+                                                       ledger::core::endianness::getSystemEndianness());
+            return result;
+        }
+
+        uint16_t BytesReader::readNextLeUint16() {
+            uint16_t result;
+            uint8_t* ptr = reinterpret_cast<uint8_t *>(&result);
+            for (auto i = 0; i < sizeof(result); i++) {
+                ptr[i] = readNextByte();
+            }
+            ledger::core::endianness::swapToEndianness(ptr, sizeof(result),
+                                                       ledger::core::endianness::Endianness::LITTLE,
+                                                       ledger::core::endianness::getSystemEndianness());
+            return result;
+        }
+
+        int16_t BytesReader::readNextBeInt16() {
+            int16_t result;
+            uint8_t* ptr = reinterpret_cast<uint8_t *>(&result);
+            for (auto i = 0; i < sizeof(result); i++) {
+                ptr[i] = readNextByte();
+            }
+            ledger::core::endianness::swapToEndianness(ptr, sizeof(result),
+                                                       ledger::core::endianness::Endianness::BIG,
+                                                       ledger::core::endianness::getSystemEndianness());
+            return result;
+        }
+
+        int16_t BytesReader::readNextLeInt16() {
+            int16_t result;
+            uint8_t* ptr = reinterpret_cast<uint8_t *>(&result);
+            for (auto i = 0; i < sizeof(result); i++) {
+                ptr[i] = readNextByte();
+            }
+            ledger::core::endianness::swapToEndianness(ptr, sizeof(result),
+                                                       ledger::core::endianness::Endianness::LITTLE,
+                                                       ledger::core::endianness::getSystemEndianness());
+            return result;
+        }
```

Comment:
There are a lot of redundant code. Please consider an alternative such as my suggestion to make the code more reusable.

```suggestion
        template <class Integer, ledger::core::endianness Endianness>
        Integer BytesReader::readNextInteger() {
            constexpr auto integerSize = sizeof(Integer);
            Integer result;
            uint8_t* ptr = reinterpret_cast<uint8_t *>(&result);
            
            for (auto i = 0; i < integerSize; i++) {
                ptr[i] = readNextByte();
            }
            
            namespace endianness = ledger::core::endianness;
            
            endianness::swapToEndianness(ptr, integerSize, Endianess, endianness::getSystemEndianness());
            return result;
        }
        
        uint16_t BytesReader::readNextBeUint16() {
            return readNextInteger<uint16_t, ledger::core::endianness::Endianness::BIG>();
        }

        uint16_t BytesReader::readNextLeUint16() {
            return readNextInteger<uint16_t, ledger::core::endianness::Endianness::LITTLE>();
        }
        
        int16_t BytesReader::readNextBeInt16() {
            return readNextInteger<int16_t, ledger::core::endianness::Endianness::BIG>();
        }

        int16_t BytesReader::readNextLeInt16() {
            return readNextInteger<int16_t, ledger::core::endianness::Endianness::LITTLE>();
        }
```